app/pb_kits/playbook/pb_source/_source.jsx

```diff
@@ -1,21 +1,129 @@
+/* @flow */
+
 import React from 'react'
-import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+import {
+  buildDataProps,
+} from '../utilities/props'
+import {
+  titleize,
+} from '../utilities/text'
+import {
+  Avatar,
+  Body,
+  Caption,
+  IconCircle,
+  Title,
+} from '../'
 
-const propTypes = {
-  className: PropTypes.string,
-  id: PropTypes.string,
+type SourceProps = {
+  className?: String,
+  data?: object,
+  hideIcon: Boolean,
+  id?: String,
+  source?: String,
+  type: "events" | "inbound" | "outbound" | "prospecting" | "referral" | "retail" | "user",
+  user: object,
 }
 
-class Source extends React.Component {
-  render() {
-    return (
-      <div className="pb_source">
-        <span>{'SOURCE CONTENT'}</span>
-      </div>
-    )
+const Source = ({
+  className,
+  data = {},
+  hideIcon = false,
+  id,
+  source,
+  type = 'inbound',
+  user = {},
+}: SourceProps) => {
+  const dataProps = buildDataProps(data)
+
+  const css = classnames([
+    'pb_source_kit',
+    className,
+  ])
+
+  const avatar = () => {
+    if ((type === 'user' || type === 'referral') && user.name !== undefined) {
+      const avatarProps = Object.assign({}, user)
```

Comment:
One this part, I'm wondering about the need for `Object.assign...`. I'm unsure if you need it as the default on `:37` should be able to cover this. If you are just cloning here, you will want to go with:

```js
const avatarProps = { ...user }
```

[reference](https://thecodebarbarian.com/object-assign-vs-object-spread.html)